invenio/legacy/search_engine/__init__.py

```diff
@@ -3952,7 +3991,18 @@ def record_public_p(recID, recreate_cache_if_needed=True):
     """Return 1 if the record is public, i.e. if it can be found in the Home collection.
        Return 0 otherwise.
     """
-    return recID in get_collection_reclist(CFG_SITE_NAME, recreate_cache_if_needed=recreate_cache_if_needed)
+    if recID in get_collection_reclist(CFG_SITE_NAME, recreate_cache_if_needed=recreate_cache_if_needed):
+        return True
+
+    if cfg.get('CFG_WEBSEARCH_PUBLIC_RECORD_EXTRA_CHECK', False):
```

Comment:
This check is disabled by default since it requires the root collection to be defined using MARC tags instead of indexes. When a record is uploaded and before it has been indexed by webcoll/bibindex, we allow to display the record in case it is in the root collection.